src/Nancy.ViewEngines.Razor.Tests/RazorViewEngineFixture.cs

```diff
@@ -648,6 +658,134 @@ public void Should_render_attributes_with_NonEncodedHtmlString_inside()
+        [InlineData(typeof(long))]
+        [InlineData(typeof(ulong))]
+        [InlineData(typeof(float))]
+        [InlineData(typeof(double))]
+        [InlineData(typeof(decimal))]
+        [InlineData(typeof(char))]
+        [InlineData(typeof(bool))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(Tuple<string>))]
+        [InlineData(typeof(Tuple<Tuple<string>>))]
+        [InlineData(typeof(Tuple<string, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int>>))]
+        [InlineData(typeof(Tuple<string, Tuple<int>, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int, string>, Tuple<int>, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int, string>, Tuple<int>, int, Tuple<string, Tuple<int, int>>>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(int[,]))]
+        [InlineData(typeof(int[, , , ,]))]
+        [InlineData(typeof(int[][]))]
+        [InlineData(typeof(int[][, ,][]))]
+        [InlineData(typeof(Tuple<int, string>[][, ,][]))]
+        [InlineData(typeof(Tuple<int, int[][, ,][], string>))]
+        public void Should_be_able_to_render_csharp_view_with_generic_model(Type expectedType)
+        {
+            this.TestGenericView(expectedType, new CSharpCodeProvider(), "model");
+        }
+
+#if !__MonoCS__
+        [Theory]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(byte))]
+        [InlineData(typeof(sbyte))]
+        [InlineData(typeof(short))]
+        [InlineData(typeof(ushort))]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(uint))]
+        [InlineData(typeof(long))]
+        [InlineData(typeof(ulong))]
+        [InlineData(typeof(float))]
+        [InlineData(typeof(double))]
+        [InlineData(typeof(decimal))]
+        [InlineData(typeof(char))]
+        [InlineData(typeof(bool))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(Tuple<string>))]
+        [InlineData(typeof(Tuple<Tuple<string>>))]
+        [InlineData(typeof(Tuple<string, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int>>))]
+        [InlineData(typeof(Tuple<string, Tuple<int>, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int, string>, Tuple<int>, int>))]
+        [InlineData(typeof(Tuple<string, Tuple<int, string>, Tuple<int>, int, Tuple<string, Tuple<int, int>>>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(int[,]))]
+        [InlineData(typeof(int[, , , ,]))]
+        [InlineData(typeof(int[][]))]
+        [InlineData(typeof(int[][, ,][]))]
+        [InlineData(typeof(Tuple<int, string>[][, ,][]))]
+        [InlineData(typeof(Tuple<int, int[][, ,][], string>))]
+        public void Should_be_able_to_render_vb_view_with_generic_model(Type expectedType)
+        {
+            this.TestGenericView(expectedType, new VBCodeProvider(), "ModelType");
+        }
+#endif
+
+        private void TestGenericView(Type expectedType, CodeDomProvider provider, string modelDirective)

```

Comment:
Lot's to say here

1. I'd rather see the `// When` and `// Then` sections moved out to the individual tests and have this method be the setup for the `// Given` section
2. See rest of the comments inline